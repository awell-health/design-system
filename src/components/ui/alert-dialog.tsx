import * as React from "react";
interface Props {
  title: string;
  text?: string;
  children?: JSX.Element | string;
  button?: {
    label: string;
    buttonAction?: () => void;
  };
}

const AlertDialog = (props: Props): React.JSX.Element => {
  const { title, text, button, children = null } = props;

  return (
    <div
      className="relative z-10"
      aria-labelledby="modal-title"
      role="dialog"
      aria-modal="true"
    >
      <div className="fixed inset-0 bg-black bg-opacity-80 transition-opacity"></div>
      <div className="fixed inset-0 z-10 w-screen overflow-y-auto">
        <div className="flex min-h-full items-end justify-center p-4 sm:items-center sm:p-0">
          <div className="relative transform overflow-hidden rounded-lg bg-white shadow-xl transition-all sm:my-8 sm:w-full sm:max-w-lg">
            <div className="bg-white p-6 pb-8">
              <div>
                <div>
                  <h3
                    className="text-base font-semibold leading-6 text-gray-900"
                    id="modal-title"
                  >
                    {title}
                  </h3>
                  {text && (
                    <div className="mt-2">
                      <p className="text-sm text-gray-500">{text}</p>
                    </div>
                  )}
                </div>
              </div>
            </div>
            {children && <div className="pb-8 px-6">{children}</div>}
            {button && (
              <div className="px-6 pb-6 flex justify-end">
                <button
                  type="button"
                  className="inline-flex justify-center rounded-md bg-blue-600 px-3 py-2 text-sm font-semibold text-white shadow-sm hover:bg-blue-500 sm:ml-3"
                  onClick={button?.buttonAction}
                >
                  {button?.label}
                </button>
              </div>
            )}
          </div>
        </div>
      </div>
    </div>
  );
};

export { AlertDialog }
