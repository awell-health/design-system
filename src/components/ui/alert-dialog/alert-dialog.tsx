import * as React from 'react';
import { cn } from '@/lib/utils';

import { Icon } from '../icon';
import { Button } from '../button/button';

interface AlertDialogButton {
  label: string;
  buttonVariant?: 'primary' | 'secondary' | 'error' | 'success' | 'warning';
  buttonAction?: () => void;
  disabled?: boolean;
}

interface Props {
  title?: string;
  icon?: JSX.Element;
  text?: string;
  children?: JSX.Element | string;
  onClose?: () => void;
  button?: AlertDialogButton;
  secondaryButton?: AlertDialogButton;
  className?: string;
}

const AlertDialog = (props: Props): React.JSX.Element => {
  const { title, icon, text, button, children = null, onClose, secondaryButton, className } = props;

  return (
    <div
      className={cn('relative z-[9999]', className)}
      aria-labelledby='modal-title'
      role='dialog'
      aria-modal='true'
    >
      <div className='fixed inset-0 bg-slate-700 bg-opacity-30 transition-opacity z-[9999]'></div>
      <div className='fixed inset-0 z-[9999] w-screen overflow-y-auto'>
        <div className='flex min-h-full items-center justify-center p-4 sm:items-center sm:p-0'>
          <div className='relative transform overflow-hidden transition-all sm:my-8 sm:w-full sm:max-w-lg p-6 bg-white rounded-lg shadow'>
            <div className='flex flex-row' style={{ gap: '20px' }}>
              {onClose && (
                <div
                  className='absolute top-0 right-0 p-4 cursor-pointer text-slate-950'
                  onClick={onClose}
                  data-testid='close-handler'
                >
                  <Icon icon='RiCloseLine' />
                </div>
              )}
              <div className='flex gap-5'>
                {icon && <div>{icon}</div>}
                <div>
                  <div className='flex flex-col'>
                    {title && (
                      <h3
                        className='text-gray-900 text-lg text-gray-900 leading-7'
                        id='modal-title'
                      >
                        {title}
                      </h3>
                    )}
                    {text && (
                      <div className='mt-2'>
                        <p className='text-gray-500 text-sm'>{text}</p>
                      </div>
                    )}
                  </div>
                </div>
              </div>
            </div>
            {children && <div>{children}</div>}
            {(button || secondaryButton) && (
              <div
                className='flex justify-end pt-[32px] gap-3 items-center'
                // FIXME: Using inline style because gap-3 is not working
                style={{ gap: '12px' }}
              >
                {secondaryButton && (
                  <Button
                    type='button'
                    variant={secondaryButton?.buttonVariant ?? 'secondary'}
                    onClick={secondaryButton?.buttonAction}
                    disabled={secondaryButton?.disabled}
                  >
                    {secondaryButton?.label}
                  </Button>
                )}
                {button && (
                  <Button
                    type='button'
                    variant={button?.buttonVariant ?? 'primary'}
                    onClick={button?.buttonAction}
                    disabled={button?.disabled}
                  >
                    {button.label}
                  </Button>
                )}
              </div>
            )}
          </div>
        </div>
      </div>
    </div>
  );
};

export { AlertDialog };
