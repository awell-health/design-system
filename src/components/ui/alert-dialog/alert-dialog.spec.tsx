import { fireEvent, render } from '@testing-library/react';
import { expect, it, describe, vi } from 'vitest';

import { AlertDialog } from './alert-dialog';

describe('Alert Dialog', () => {
  const subject = (props = {}) => {
    const defaultProps = {
      title: 'title',
      icon: <span>icon</span>,
      text: 'text',
      children: <span>children</span>
    };

    return render(<AlertDialog {...defaultProps} {...props} />);
  };

  it('match snapshot', () => {
    expect(subject()).toMatchSnapshot();
  });

  it('handles onClose', () => {
    const onCloseMock = vi.fn();
    const { getByTestId } = render(<AlertDialog onClose={onCloseMock} />);

    fireEvent.click(getByTestId('close-handler'));

    expect(onCloseMock).toHaveBeenCalledOnce();
  });
});
