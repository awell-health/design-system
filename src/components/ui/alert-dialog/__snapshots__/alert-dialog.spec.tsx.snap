// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Alert Dialog > match snapshot 1`] = `
<div>
  <div
    aria-labelledby="modal-title"
    aria-modal="true"
    class="relative z-[9999]"
    role="dialog"
  >
    <div
      class="fixed inset-0 bg-slate-700 bg-opacity-30 transition-opacity z-[9999]"
    />
    <div
      class="fixed inset-0 z-[9999] w-screen overflow-y-auto"
    >
      <div
        class="flex min-h-full items-center justify-center p-4 sm:items-center sm:p-0"
      >
        <div
          class="relative transform overflow-hidden transition-all sm:my-8 sm:w-full sm:max-w-lg p-6 bg-white rounded-lg shadow"
        >
          <div
            class="flex flex-row"
            style="gap: 20px;"
          >
            <div
              class="flex gap-5"
            >
              <div>
                <span>
                  icon
                </span>
              </div>
              <div>
                <div
                  class="flex flex-col"
                >
                  <h3
                    class="text-gray-900 text-lg text-gray-900 leading-7"
                    id="modal-title"
                  >
                    title
                  </h3>
                  <div
                    class="mt-2"
                  >
                    <p
                      class="text-gray-500 text-sm"
                    >
                      text
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div>
            <span>
              children
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
